frontend/public/storage/components/storage-overview/storage-overview.jsx

```diff
@@ -1,16 +1,133 @@
 import React from 'react';
+import {
+  StorageOverview as KubevirtStorageOverview,
+  StorageOverviewContext,
+  getResource,
+} from 'kubevirt-web-ui-components';
 
-import * as dashboardMockup from '../../../imgs/ocs-dashboard.png';
+import {
+  CephClusterModel,
+  NodeModel,
+  PodModel,
+  PersistentVolumeClaimModel,
+  PersistentVolumeModel,
+  VirtualMachineModel,
+  VirtualMachineInstanceMigrationModel,
+} from '../../../models';
+
+import { WithResources } from '../../../kubevirt/components/utils/withResources';
+import { LoadingInline } from '../../../kubevirt/components/utils/okdutils';
+import { coFetchJSON } from '../../../co-fetch';
+
+const REFRESH_TIMEOUT = 30000;
+const CEPH_ROOK_NAMESPACE = 'openshift-storage';
+
+const CEPH_STATUS = 'ceph_health_status';
+
+const resourceMap = {
+  nodes: {
+    resource: getResource(NodeModel, { namespaced: false }),
+  },
+  pods: {
+    resource: getResource(PodModel, {
+      namespace: CEPH_ROOK_NAMESPACE,
+    }),
+  },
+  pvs: {
+    resource: getResource(PersistentVolumeModel),
+  },
+  pvcs: {
+    resource: getResource(PersistentVolumeClaimModel),
+  },
+  vms: {
+    resource: getResource(VirtualMachineModel),
+  },
+  migrations: {
+    resource: getResource(VirtualMachineInstanceMigrationModel),
+  },
+  cephCluster: {
+    resource: getResource(CephClusterModel, { isList: false }),
+  },
+};
 
 export class StorageOverview extends React.Component {
-  constructor(props){
+  constructor(props) {
     super(props);
+    this.state = {
+      ocsHealthData: {
+        data: {},
+        loaded: false,
+      },
+    };
+    this.setHealthData = this._setHealthData.bind(this);
+  }
+  _setHealthData(healthy) {
+    this.setState({
+      ocsHealthData: {
+        data: {
+          healthy,
+        },
+        loaded: true,
+      },
+    });
+  }
+
+  fetchHealth(response, callback) {
+    const result = response.data.result;
```

Comment:
it would be better if you pass the whole response to `Health` component and it would contain logic how to parse the `response` data. You should also start thinking about how to handle cases when prometheus is not available (pod may not be running or not created at all.. ) See https://github.com/kubevirt/web-ui-components/issues/310

But that can be done as a followup IMHO